### PR TITLE
142652743 : Unit test failing for pruner sometimes

### PR DESCRIPTION
**Pull request by ghaih**

### CODE REVIEW

**Comment by kanekoh** [comment]
LGTM

File: src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go [hunk 1]
```diff
@@ -66,16 +66,17 @@ func initConsul() {
 
 func initConfig() {
 	cfg.Logging.Level = "debug"
+	dbUrl := os.Getenv("DBURL")
```
Comment (boyang9527) [inline src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go]: fail the test if $DBURL is not set

File: src/autoscaler/pruner/cmd/pruner/pruner_test.go [hunk 2]
```diff
@@ -73,9 +73,19 @@ var _ = Describe("Pruner", func() {
 				}))
 		})
 
-		It("should start", func() {
```
Comment (boyang9527) [inline src/autoscaler/pruner/cmd/pruner/pruner_test.go]: if you separate into multiple test cases, then you don't need duplicate test of 
Eventually(runner.Session.Buffer, 2*time.Second).Should(Say("pruner.started")) Consistently(runner.Session).ShouldNot(Exit())

You can have another test case to test for "prunder started"

File: src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go (hunk 1 above)
Comment (ghaih) [inline src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go]: It was moved as a part of refactoring as the same code was repeated. The sqldb_suite_test, covers the $DBURL not set case. Am I missing something. As os.Getenv("DBURL") is being used all over the place.

File: src/autoscaler/pruner/cmd/pruner/pruner_test.go (hunk 2 above)
Comment (ghaih) [inline src/autoscaler/pruner/cmd/pruner/pruner_test.go]: Yes can separate test for .Should(Say("pruner.started")) but still need eachConsistently(runner.Session).ShouldNot(Exit()) in each test isn't it

File: src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go (hunk 1 above)
Comment (boyang9527) [inline src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go]: Perhaps I missed something... Pruner test cases can be run separately or before db package test cases. 

we can just add check before the following line and fail the test of $DBURL is not set
https://github.com/cloudfoundry-incubator/app-autoscaler/blob/develop/src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go#L70

File: src/autoscaler/pruner/cmd/pruner/pruner_test.go (hunk 2 above)
Comment (boyang9527) [inline src/autoscaler/pruner/cmd/pruner/pruner_test.go]: yes, you are right

File: src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go (hunk 1 above)
Comment (ghaih) [inline src/autoscaler/pruner/cmd/pruner/pruner_suite_test.go]: Fixed for pruner, just a note, similar usage in other components.

File: src/autoscaler/pruner/cmd/pruner/pruner_test.go (hunk 2 above)
Comment (ghaih) [inline src/autoscaler/pruner/cmd/pruner/pruner_test.go]: Fixed

Comment (boyang9527) [inline src/autoscaler/pruner/cmd/pruner/pruner_test.go]: thanks!

**Comment by boyang9527** [comment]
LGTM

**Pull request merged**